track.c

```diff
@@ -250,6 +250,57 @@ void sorted_list_remove_track(struct list_head *head, struct rb_root *tree_root,
 	list_del(&track->node);
 }
 
+static void shuffle_array(void *array, size_t n, size_t size) 
+{
+	char tmp[size];
+	char *arr = array;
+	size_t stride = size * sizeof(char);
+	if (n > 1) {
+		size_t i;
+		for (i = 0; i < n - 1; ++i) {
```

Comment:
These 3 lines can be replaced by something like `for (ssize_t i = 0; i < (ssize_t)n - 1; ++i)`.